### ISSUE TITLE
plotly loses selection when toggling app view from edit view

### ISSUE THREAD

**Issue by alefminus** [marimo-team/marimo#1160]
### Describe the bug

When a plotly figure has a selection that selection is erased upon switching to app view

[Screencast from 2024-04-18 09-33-22.webm](https://github.com/marimo-team/marimo/assets/103570620/7d02c0a1-fcd4-4d38-81c3-dc8f9b6416bf)

### Environment

```json
{
  "marimo": "0.4.0",
  "OS": "Linux",
  "OS Version": "6.8.6-200.fc39.x86_64",
  "Processor": "",
  "Python Version": "3.12.2",
  "Binaries": {
    "Browser": "--",
    "Node": "v20.11.1"
  },
  "Requirements": {
    "click": "8.1.3",
    "importlib-resources": "missing",
    "jedi": "0.19.1",
    "markdown": "3.5.2",
    "pymdown-extensions": "10.7.1",
    "pygments": "2.15.1",
    "tomlkit": "0.12.4",
    "uvicorn": "0.28.0",
    "starlette": "0.37.2",
    "websocket": "missing",
    "typing-extensions": "4.10.0",
    "black": "24.1.1"
  }
}
```

### Code to reproduce

```python
import marimo

__generated_with = "0.4.0"
app = marimo.App()


@app.cell
def __():
    import marimo as mo
    import plotly.express as px
    return mo, px


@app.cell
def __(mo, px):
    mo.ui.plotly(px.scatter(x=[1, 2, 3, 4, 5, 6], y=[1, 2, 3, 2, 3, 4]))
    return


@app.cell
def __():
    return


if __name__ == "__main__":
    app.run()
```

**Comment by mscolnick** [comment]
I'll look into this today

**Comment by mscolnick** [comment]
@alefminus - thanks for debugging that. I will fix that

**Comment by mscolnick** [comment]
@alefminus this should be fixed now on `0.4.3`. going to close, but let me know if you encounter other related issues
